app/assets/javascripts/helpers/toMoment.js

```diff
@@ -0,0 +1,28 @@
+// Allow a few strict formats of dates
+const allowedFormats = [
+  'MM/DD/YYYY',
+  'MM/D/YYYY',
+  'M/DD/YYYY',
+  'M/D/YYYY',
+
+  'MM/DD/YY',
+  'MM/D/YY',
+  'M/DD/YY',
+  'M/D/YY',
+
+  'MM-DD-YYYY',
+  'MM-D-YYYY',
+  'M-DD-YYYY',
+  'M-D-YYYY',
+
+  'MM-DD-YY',
+  'MM-D-YY',
+  'M-DD-YY',
+  'M-D-YY'
+];
+
+// This will always return a `moment` but if`dateText` can't be parsed, that moment will return false
+// when you call `isValid`.
+export function toMoment(dateText) {
+  return moment.utc(dateText, allowedFormats, true); // this does 'strict parsing' on multiple formats
```

Comment:
Learning about some great features of `moment` here, if other folks are reviewing and want to follow along here are the moment docs for the features @kevinrobinson is using:

+ https://momentjs.com/guides/#/parsing/strict-mode/
+ https://momentjs.com/guides/#/parsing/multiple-formats/

@kevinrobinson, the multiple formats doc mentions that "the more formats that are used, the longer that parsing takes." We should check out the impact on the client side performance, I don't imagine it will be huge but just something I'll think about when I test out in IE.

Reply:
Tested this out, adding these format's didn't make service saving slow. 👍

Reply:
@alexsoble Yeah, good to check.  I couldn't notice either  👍